### PR TITLE
Kernel 5.10

### PR DESCRIPTION
**Pull request by patricklauer**
plus eapi7
plus slot per major kernel version, which should simplify updates

### CODE REVIEW

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	sys-apps/fakeroot
+	"
+
+REQUIRED_USE="
+	protection? ( !docker )
+	docker? ( !protection )
+"
+
+src_unpack() {
+	mkdir -p "$S"
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:33]: `|| die`

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+src_prepare() {
+	mkdir -p "$S/tmp/kernel"
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:37]: Ditto for this line and the 2 lines below.

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	export KBUILD_OUTPUT="$S/tmp/kernel"
+
+	# hnurgh.
+	export KERNEL_OUTPUTDIR="$S/tmp/kernel"
+
+	# make runs from src dir with output going to KBUILD_OUTPUT
+	# hacky PVR special
+	cd /usr/src/linux-${PV}-gentoo || die
+	make -s oldconfig || die
+	echo "Building kernel ..."
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:63]: `einfo` ?

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		--kernel-outputdir="$S/tmp/kernel/" \
+		--cachedir="$S/cache" \
+		--tmpdir="$S/tmp" \
+		--initramfs-filename='initramfs-genkernel-x86_64-%%KV%%' \
+		--loglevel=5  --no-sandbox || die
+}
+
+src_install() {
+	# don't package firmware files, provided by linux-firmware if required
+	rm -rf "$S/final/lib/firmware" || die
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:91]: `rm` always return 0 with the `-f` option. Consider removing it.

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	cp "$S/tmp/kernel/Module.symvers" "$S/final/Module.symvers"
+	cd "$S/final/" && tar cJf binkernel-${PV}.tar.xz * || die
+	mkdir -p "${D}/usr/share"
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:94]: `|| die`

File: sys-kernel/buildkernel/buildkernel-5.9.15.ebuild [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2020 Gentoo Authors
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.9.15.ebuild:1]: Same comments apply for this ebuild.

File: sys-kernel/gentoo-kernel/gentoo-kernel-5.10.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+"
+
+S=${WORKDIR}
+
+src_compile() {
+	:;
+}
+
+src_install() {
+	cp -var "${WORKDIR}"/* "${D}"
```
Comment (monsieurp) [inline sys-kernel/gentoo-kernel/gentoo-kernel-5.10.1.ebuild:54]: add `|| die` here and to all commands below.

File: sys-kernel/gentoo-kernel/gentoo-kernel-5.10.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+		cp "${D}/Module.symvers" "${D}"/usr/src/linux-${PV}-gentoo/Module.symvers
+	fi
+	rm "${D}"/Module.symvers
+}
+
+pkg_postinst() {
+	if use sources; then
+		unset ARCH
+		einfo "Preparing kernel sources"
+		cd /usr/src/linux-${PV}-gentoo/ && make -s modules_prepare
```
Comment (monsieurp) [inline sys-kernel/gentoo-kernel/gentoo-kernel-5.10.1.ebuild:73]: Break it down into two seperate commands and also add `|| die`.

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+		--cachedir="$S/cache" \
+		--tmpdir="$S/tmp" \
+		--initramfs-filename='initramfs-genkernel-x86_64-%%KV%%' \
+		--loglevel=5  --no-sandbox || die
+}
+
+src_install() {
+	# don't package firmware files, provided by linux-firmware if required
+	rm -rf "$S/final/lib/firmware" || die
+	cp "$S/tmp/kernel/Module.symvers" "$S/final/Module.symvers"
```
Comment (monsieurp) [inline sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:92]: `|| die`

**Review by monsieurp — CHANGES_REQUESTED** [review]
Final comment: the commit message should be `$CAT/$PN` but you know that already :)

File: sys-kernel/buildkernel/buildkernel-5.10.1.ebuild (hunk 1 above)
Comment (patricklauer) [reply sys-kernel/buildkernel/buildkernel-5.10.1.ebuild:33]: Can't fail, so why care ;)

**Pull request merged**